### PR TITLE
Update the docs for V2 Changes

### PR DESCRIPTION
**Pull request by cloneofghosts**
Update the docs to be fully updated for V2 and finish HA documentation

- Fix https://github.com/Pirate-Weather/pirateweather/issues/214
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/146
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/211
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/231
- Fix https://github.com/Pirate-Weather/pirate-weather-ha/issues/213
- Fix https://github.com/Pirate-Weather/pirateweather/issues/215

### CODE REVIEW

**Comment by alexander0042** [comment]
Great, just pushed another set of additions to the V2.0 docs branch, which should round out all those updates. 

One thing I was thinking about was what sort of stuff should go in the HA docs that isn't already included? I added some stuff on alerts, but are there more FAQ's that should be added? Or something more fundamental?

**Comment by cloneofghosts** [comment]
Was looking through the changes and I don't see anything specific in the HA docs about the alerts sensor. Maybe I missed it but it looks like it's not there? For the HA docs I'd include a link to the data points section in the API documentation. I would document any specific HA things in there that don't apply to the regular API. Be it as FAQs or just regular documentation.

The only other thing that is needed here is updating the OpenAPI specification as it looks like it was missed in these changes.

File: docs/API.md [hunk 1]
```diff
@@ -472,25 +460,25 @@ The time of the minimum "feels like" temperature during the daytime, from 6:00 a
...
-**Only on `daily`.** The maxiumum `fireIndex` for the given day.
+**Only on `daily`.** The maximum `fireIndex` for the given day.
 
 #### fireIndexMaxTime
-**Only on `daily`.** the time in which the maxiumum `fireIndex` occurs represented in UNIX time.
+**Only on `daily`.** the time in which the maximum `fireIndex` occurs represented in UNIX time.
 
 #### humidity
-Relative humidity expressed as a value between 0 and 1 inclusive. This is a percentage of the actual water vapor in the air compared to the total amount of water vapor that can exist at the current temperature. [See this resource for more information.](https://www.sciencedirect.com/topics/agricultural-and-biological-sciences/relative-humidity)
+Relative humidity expressed as a value between 0 and 1 inclusive. This is a percentage of the actual water vapor in the air compared to the total amount of water vapour that can exist at the current temperature. [See this resource for more information.](https://www.sciencedirect.com/topics/agricultural-and-biological-sciences/relative-humidity)
```
Comment (cloneofghosts) [inline docs/API.md:481]: There is another instance of vapor that needs changed here

File: docs/DataSources.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# API Docs
+This page serves as the documentation for the underlying data source algorithm for the Pirate Weather API- in sort, it explains which parameter comes from where. Since the goal of this API to to provide raw model data with as little processing as possible, results from the API should very closely match the model described in this document, with some minor differences due to interpolation. 
+
+## Data sources
+Several models are used to produce the forecast. They are all hosted on [AWS's Open Data Platform](https://registry.opendata.aws/collab/noaa/), and the fantastic [Herbie package](https://github.com/blaylockbk/Herbie) is used to download and perform initial processing for all of them.    
+
+#### NBM
+The National Blend of Models [(NBM)](https://vlab.noaa.gov/web/mdl/nbm) is a calibrated blend of both NOAA and non-NOAA weather models from around the world. Running every hour for about days, the NBM produces a forecast that aims to leverage strengths from each of the source models, as well as providing some probabilistic forecasts. For most weather elements in the US and Canada, this is the primary source.
```
Comment (cloneofghosts) [inline docs/DataSources.md:8]: Missing the number of days

**Comment by cloneofghosts** [comment]
Also as per [this comment](https://github.com/Pirate-Weather/pirate-weather-ha/issues/230#issuecomment-2097050507) maybe you should clarify where the storm distance is measured from?

File: docs/DataSources.md [hunk 3]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
 Every Pirate Weather forecast element for each time block (`currently`, `minutely`, `hourly`, or `daily`) is included in the table below, along with the primary, secondary, and tertiary data sources. Fallback sources are used if model data is intentionally excluded, the request point is outside of the primary model coverage area, or if there's some sort of data interruption. 
 
 
-|Parameter |Currently   |Minutely   |Hourly   |Daily   |
+|Parameter 	|Currently   |Minutely   |Hourly/ Daily   |
 |---|---|---|---|---|
```
Comment (cloneofghosts) [inline docs/DataSources.md:30]: Remove extra row to fix table formatting

File: docs/DataSources.md [hunk 4]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|dewPoint     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|fireIndex    			|NBM   			  		|N/A   	|NBM   			 			|
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
+|ozone   				|GFS   					|N/A   	|GFS   						|
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|precipIntensity   		|NBM > HRRR > GEFS 		|HRRR	|NBM > HRRR > GEFS			|
```
Comment (cloneofghosts) [inline docs/DataSources.md:42]: Shouldn't this be HRRR > NBM > GEFS for all sections?

File: docs/DataSources.md [hunk 5]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
-#### 
-
-#### windSpeed
-
+|apparentTemperature	|NBM > GFS				|N/A   	|NBM > GFS   			 	|
+|cloudCover   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|		
+|dewPoint     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|fireIndex    			|NBM   			  		|N/A   	|NBM   			 			|
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
```
Comment (cloneofghosts) [inline docs/DataSources.md:36]: Shouldn't the currently section be N/A? There is no `iceAccumulaton` field on the currently section.

File: docs/DataSources.md [hunk 6]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
```
Comment (cloneofghosts) [inline docs/DataSources.md:37]: Shouldn't the currently section be N/A? There is no `liquidAccumulaton` field on the currently section.

File: docs/DataSources.md [hunk 7]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
```
Comment (cloneofghosts) [inline docs/DataSources.md:38]: Shouldn't it be HRRR > GFS? Is NBM data used at all?

File: docs/DataSources.md [hunk 8]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
```
Comment (cloneofghosts) [inline docs/DataSources.md:39]: Shouldn't it be HRRR > GFS? Is NBM data used at all?

File: docs/DataSources.md [hunk 9]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|ozone   				|GFS   					|N/A   	|GFS   						|
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|precipIntensity   		|NBM > HRRR > GEFS 		|HRRR	|NBM > HRRR > GEFS			|
+|precipIntensityError	|GEFS					|GEFS	|GEFS						|	
+|precipProbability  	|NBM > GEFS 			|NBM 	|NBM > GEFS					|
+|precipType   			|NBM > HRRR > GEFS 		|HRRR  	|NBM > HRRR > GEFS			|
```
Comment (cloneofghosts) [inline docs/DataSources.md:45]: Shouldn't the currently section be HRRR > NBM > GEFS  and the minutely section HRRR > NBM > GEFS?

File: docs/DataSources.md [hunk 10]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
+|ozone   				|GFS   					|N/A   	|GFS   						|
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|precipIntensity   		|NBM > HRRR > GEFS 		|HRRR	|NBM > HRRR > GEFS			|
+|precipIntensityError	|GEFS					|GEFS	|GEFS						|	
+|precipProbability  	|NBM > GEFS 			|NBM 	|NBM > GEFS					|
```
Comment (cloneofghosts) [inline docs/DataSources.md:44]: Shouldn't this always be NBM > GEFS?

File: docs/DataSources.md [hunk 11]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|precipType   			|NBM > HRRR > GEFS 		|HRRR  	|NBM > HRRR > GEFS			|
+|pressure   			|HRRR > GFS   			|N/A	|HRRR > GFS 				|
+|snowAccumulation   	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS 	|
```
Comment (cloneofghosts) [inline docs/DataSources.md:47]: Shouldn't the currently section be N/A? There is no `snowAccumulaton` field on the currently section.

File: docs/DataSources.md [hunk 12]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|smoke   				|NBM   					|N/A   	|NBM   						|
```
Comment (cloneofghosts) [inline docs/DataSources.md:48]: Should be HRRR and not NBM

File: docs/DataSources.md [hunk 13]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|temperature   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Comment (cloneofghosts) [inline docs/DataSources.md:49]: Shouldn't the currently section be HRRR > NBM > GFS?

File: docs/DataSources.md [hunk 14]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|uvIndex   				|GFS   					|N/A   	|GFS   						|
+|visibility   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|windBearing  			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Comment (cloneofghosts) [inline docs/DataSources.md:52]: Shouldn't the currently section be HRRR > NBM > GFS?

File: docs/DataSources.md [hunk 15]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|windGust   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Comment (cloneofghosts) [inline docs/DataSources.md:53]: Shouldn't the currently section be HRRR > NBM > GFS?

File: docs/DataSources.md [hunk 16]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|windSpeed   			|NBM > HRRR > GFS   	|N/A	|NBM > HRRR > GFS   		|
```
Comment (cloneofghosts) [inline docs/DataSources.md:54]: Shouldn't the currently section be HRRR > NBM > GFS?

File: docs/DataSources.md [hunk 17]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
+|cloudCover   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|		
+|dewPoint     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|fireIndex    			|NBM   			  		|N/A   	|NBM   			 			|
+|humidity     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
+|iceAccumulation   		|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|liquidAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
+|nearestStormBearing	|GFS   					|N/A   	|GFS   						|
+|nearestStormDistance   |GFS   					|N/A   	|GFS   						|
+|ozone   				|GFS   					|N/A   	|GFS   						|
+|precipAccumulation 	|NBM > HRRR > GEFS > GFS|N/A   	|NBM > HRRR > GEFS > GFS	|
```
Comment (cloneofghosts) [inline docs/DataSources.md:41]: Shouldn't the currently section be N/A? There is no `precipAccumulaton` field on the currently section.

File: docs/DataSources.md [hunk 18]
```diff
@@ -26,64 +26,29 @@ To provide historic weather data, the [European Reanalysis 5 Dataset](https://re
...
-
-#### 
-
-#### 
-
-#### windSpeed
-
+|apparentTemperature	|NBM > GFS				|N/A   	|NBM > GFS   			 	|
+|cloudCover   			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|		
+|dewPoint     			|NBM > HRRR > GFS   	|N/A   	|NBM > HRRR > GFS   		|
```
Comment (cloneofghosts) [inline docs/DataSources.md:33]: Shouldn't the currently section be HRRR > NBM > GFS?

**Review by cloneofghosts — COMMENTED** [review]
Just had another look through the data sources document and just wanted to confirm where the data is coming from. Thought that the comment would just show the line I added the comment to. If you click view reviewed changes it will show the comment better,

**Comment by alexander0042** [comment]
Incredible notes, these are great! I'll go through them and double check each one.

Also, note for myself- smoke is ins kg/m3 (the model default), but ug/m3 makes way more sense. Need the docs to report ug/m3 and multiply the response by 10^9

**Comment by cloneofghosts** [comment]
I fixed the missed spelling change and I updated a few items in the table that I'm sure of but you should double check and make sure.

File: docs/DataSources.md (hunk 7 above)
Comment (alexander0042) [reply docs/DataSources.md:38]: Storm bearing is always 100% GFS for now- it's a pretty heavy computational lift to calculate these two, and so it's much easier to just do it for one model every 6-hours

Comment (cloneofghosts) [reply docs/DataSources.md:38]: Okay good to know. I remember seeing a in the nearest storm issue that you were going to use HRRR but it must have changed at some point.

**Comment by alexander0042** [comment]
Like I said earlier, amazing edits!!! I went through and put them all in, as well as updating the data pipeline page with the new times. I'm going to supersede that page with the details on the new V2 processing flow, but since that's just for general interest I want to get this out first.

Also added some more details on the storm distance, and updated the changelog/ roadmap. 

Also really good comment on the OpenAPI specs. Apiable really likes having those, and it's good practice, so I'll add those in before we push this.

**Comment by cloneofghosts** [comment]
@alexander0042 I added in the new sensors to the HA page and did some minor updates to the OpenAPI spec (mainly change data types on some items which were incorrect). Also the DataSources.md page will need to be added to the menu.

**Comment by cloneofghosts** [comment]
@alexander0042 I've updated the OpenAPI spec if you want to double check and make sure everything is good there. I also added in the new sensors into the HA docs page. Is there anything from my last comment that you want to add or is it good to merge?

**Comment by cloneofghosts** [comment]
I added the new sensors which I added yesterday and the template from the templating issue in HA. @alexander0042 I think things are good now if you could review so we can get this merged in.

**Comment by alexander0042** [comment]
Ok,  with that fire out I finally had a chance to review this! Read through it all and it's looking good, so I'm going to push to main and see how everything formats

**Review by alexander0042 — COMMENTED** [review]
Great edits, looks good to me!

**Comment by cloneofghosts** [comment]
Was just about to go and fix the changelog version but you beat me to it and added 2.0.6 and 2.0.7. The OpenAPI spec will need to have the versions changed but that's pretty minor.

**Pull request merged**